src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -343,7 +343,7 @@ class column_store
                 // replacements either start before or exactly at 0
                 return size_t{0};
             }
-            auto candidate = _base_offset.find(offset_seg_it->first());
+            auto candidate = _base_offset.lower_bound(offset_seg_it->first());
             if (candidate.is_end()) {
```

Comment:
I'm kind of surprised there aren't more cases handled here. If `lower_bound` returned a non-`end()` iterator then its either a direct hit (like `find` before) or the found `_base_offset` is greater than what we searched for so the end of the segments would need to be compared to identify overlap.

And if `lower_bound` returned `end()` then the base offset of `std::prev(end())` is less than what we searched for, which means there still may be an overlap.

Granted I don't really know what is happening in the `end()` and non-`end()` cases, so it may be fine, but it also seems like there could be some subtle cases occurring here.